src/tools/rbd_mirror/image_replayer/journal/Replayer.cc

```diff
@@ -1270,8 +1272,16 @@ void Replayer<I>::register_perf_counters() {
 
   auto cct = static_cast<CephContext *>(m_state_builder->local_image_ctx->cct);
   auto prio = cct->_conf.get_val<int64_t>("rbd_mirror_image_perf_stats_prio");
-  PerfCountersBuilder plb(g_ceph_context, "rbd_mirror_image_" + m_image_spec,
-                          l_rbd_mirror_journal_first, l_rbd_mirror_journal_last);
+
+  auto *local_image_ctx = m_state_builder->local_image_ctx;
+  std::string labels = ceph::perf_counters::key_create(
+      "rbd_mirror_image",
+      {{"pool", local_image_ctx->md_ctx.get_pool_name()},
+       {"namespace", local_image_ctx->md_ctx.get_namespace()},
+       {"image", local_image_ctx->name}});
+
+  PerfCountersBuilder plb(g_ceph_context, labels, l_rbd_mirror_journal_first,
+                          l_rbd_mirror_journal_last);
   plb.add_u64_counter(l_rbd_mirror_replay, "replay", "Replays", "r", prio);
   plb.add_u64_counter(l_rbd_mirror_replay_bytes, "replay_bytes",
                       "Replayed data", "rb", prio, unit_t(UNIT_BYTES));
```

Comment:
Again, let's take advantage of the fact that backwards compatibility is being broken and add a prefix and better descriptions:
```
  plb.add_u64_counter(l_rbd_mirror_journal_entries, "entries",
                      "Number of entries replayed", nullptr, prio);
  plb.add_u64_counter(l_rbd_mirror_journal_replay_bytes, "replay_bytes",
                      "Total bytes replayed", nullptr, prio,
                      unit_t(UNIT_BYTES));
  plb.add_time_avg(l_rbd_mirror_journal_replay_latency, "replay_latency",
                   "Replay latency", nullptr, prio);
```